crates/ditto-codegen-js/src/convert.rs

```diff
@@ -504,23 +542,94 @@ fn convert_effect(
         } => Block::ConstAssignment {
             ident: Ident::from(name),
             value: Expression::Call {
-                function: Box::new(convert_expression(imported_module_idents, expression)),
+                function: Box::new(convert_expression(
+                    supply,
+                    imported_module_idents,
+                    expression,
+                )),
                 arguments: vec![],
             },
-            rest: Box::new(convert_effect(imported_module_idents, rest)),
+            rest: Box::new(convert_effect(supply, imported_module_idents, rest)),
+        },
+        ditto_ast::Effect::Let {
+            pattern: ditto_ast::Pattern::Variable { name, .. },
+            box expression,
+            box rest,
+        } => Block::ConstAssignment {
+            ident: Ident::from(name),
+            value: convert_expression(supply, imported_module_idents, expression),
+            rest: Box::new(convert_effect(supply, imported_module_idents, rest)),
         },
+        ditto_ast::Effect::Let {
+            pattern: ditto_ast::Pattern::Unused { unused_name, .. },
+            box expression,
+            box rest,
+        } => {
+            // REVIEW: could just drop the unused assignment altogether?
+            Block::ConstAssignment {
+                ident: Ident::from(unused_name),
```

Comment:
This _might_ blowup if we have something like...

```ditto
let _ = 1;
let _ = 2;
let _ = 3;
```

In which case we could end generating dodgey `const` assignments, i.e.

```javascript
const _ = 1;
const _ = 2;  // Identifier '_' has already been declared
const _ = 3;
```